x/emissions/keeper/keeper.go

```diff
@@ -3996,17 +3997,44 @@ func (k *Keeper) PruneReputerNonces(ctx context.Context, topicId uint64, blockHe
 }
 
 // Return true if the nonce is within the worker submission window for the topic
-// Inclusive of the start block height and exclusive of the end block height
-func (k *Keeper) BlockWithinWorkerSubmissionWindowOfNonce(topic types.Topic, nonce types.Nonce, blockHeight int64) bool {
-	return nonce.BlockHeight <= blockHeight && blockHeight < topic.WorkerSubmissionWindow+nonce.BlockHeight
+// Inclusive of the start block height and of the end block height.
+func (k *Keeper) BlockWithinWorkerSubmissionWindowOfNonce(topic types.Topic, nonce types.Nonce, blockHeight int64) (bool, error) {
+	if topic.WorkerSubmissionWindow == 0 {
+		return false, errorsmod.Wrap(types.ErrInvalidValue, "worker submission window cannot be 0")
+	}
+	lowerBound := nonce.BlockHeight
+	upperBound := nonce.BlockHeight + topic.WorkerSubmissionWindow
+	if math.MaxInt64-lowerBound < topic.WorkerSubmissionWindow {
+		return false, errorsmod.Wrapf(types.ErrInvalidValue,
+			"nonce block height %d is too high, adding window %d would overflow",
+			nonce.BlockHeight,
+			topic.WorkerSubmissionWindow)
+	}
```

Comment:
```suggestion
	lowerBound := nonce.BlockHeight
	if math.MaxInt64-lowerBound < topic.WorkerSubmissionWindow {
		return false, errorsmod.Wrapf(types.ErrInvalidValue,
			"nonce block height %d is too high, adding window %d would overflow",
			nonce.BlockHeight,
			topic.WorkerSubmissionWindow)
	}
	upperBound := nonce.BlockHeight + topic.WorkerSubmissionWindow
```

It would overflow before checking for it :)
BTW do you think this is worth checking? If we manage `nonce.BlockHeight` only internally we should not have problems before billions of years, and then it won't be our problem anymore 😉

Reply:
I had my doubts as well but I normally prefer to bulletproof. I was thinking in potential malicious topic definitions.

Reply:
Better to be bulletproof than the opposite :),  and yeah if see that we don't restrict values on window related configs so better to check these